ext/sockets/sockets.c

```diff
@@ -1078,8 +1079,8 @@ PHP_FUNCTION(socket_getsockname)
 			break;
 
 		default:
-			php_error_docref(NULL, E_WARNING, "Unsupported address family %d", sa->sa_family);
-			RETURN_FALSE;
+			zend_argument_value_error(1, "socket address family must be AF_UNIX, AF_INET, or AF_INET6");
```

Comment:
I think the `socket address family` text won't work well with an argument error this way. So the message could rather be `must be either an AF_UNIX, AF_INET, or AF_INET6 socket`, or something similar. :)